### ISSUE TITLE
Message timing and ordering on web talk

### ISSUE THREAD

**Issue by mattlevan** [urbit/arvo#484]
![talk-times](https://user-images.githubusercontent.com/4582375/33843298-d7a446f6-de51-11e7-9b9c-69ad2f902ebc.png)

The messages are not printed in order according to their times.

**Comment by Fang-** [comment]
Fairly sure it's normal for them to be printed based on receive-order, not on timestamp. I'm seeing this in the cli as well (`;set showtime`) for the messages you listed.

What's happening here is that messages get created by the sender, including the timestamp. If two participants' system clocks are out of sync, you can get scenarios like these.

I think the sane solution here is to allow the host to set the timestamp, if the sender so desires. We might want to make a stronger distinction between "I want to send this" and "this has been sent" message types. That might tie nicely into adding delivery information back in. I'll have to sit on this for a while though.

So, not really a bug, but I'll keep this open as its undesirable behavior in some cases.

**Comment by mattlevan** [comment]
Good point. Most users expect to have a new message (just received) appear as the most recent message in their feed, even if it had been sent eons ago! Anywho, just an observation here, not a pressing issue!

**Comment by galenwp** [comment]
We used to use timestamp for message ordering, but because of network latency the behavior was a bit weird. That is, you'd have old messages appearing in your backlog and wouldn't see them. 

So, instead, we order by received. Sometimes stale messages appear late — but they're visually obvious. Open to changing this. Or to forcing everyone to have an atomic clock on their computer.

**Comment by mattlevan** [comment]
@galenwp I think it's fine as it is. The only other suitable option depends on web talk notifications. If those were implemented, you could order the messages by time and notify the user when an old one (by timestamp) finally arrived.
